### ISSUE TITLE
LiveHdtDatasource tests intermittently finalize before the queries have run

### ISSUE THREAD

**Issue by pabloem** [pabloem/Server.js#8]
I am not quite sure how to use the framework. I have an asynchronous function that calls `itShouldExecute`, but the tests often finalize -and succeed- before the queries have been run and the results verified.
Where did you read up on testing asynchronous stuff, @RubenVerborgh? : )

**Comment by RubenVerborgh** [comment]
Can you give an example of such a test? I suspect you are not providing a `done` argument.

**Comment by pabloem** [comment]
The set of tests starting in [LiveHdtDatasource.js:220](https://github.com/pabloem/Server.js/blob/offset_cache/test/datasources/LiveHdtDatasource-test.js#L220).
Specifically, The tests [run updates first](https://github.com/pabloem/Server.js/blob/offset_cache/test/datasources/LiveHdtDatasource-test.js#L259), and once the updates have succeeded, run the [actual tests](https://github.com/pabloem/Server.js/blob/offset_cache/test/datasources/LiveHdtDatasource-test.js#L239).
I guess that since the function `afterUpdates` runs asynchronously, mocha doesn't 'wait' for it to run. So sometimes it runs and sometimes it doesnt.
I read somewhere that we can use 'promises' for this sort of thing, but I haven't found useful documentation.

**Comment by RubenVerborgh** [comment]
I would expect:

```
it('should not show any removed triples, and show all added',function(done) {
```

and then `done` being called after all is finished.

But in general, your test is doing way too much. Each `itShouldExecute` should be a separate `it` under a `describe`.

**Comment by RubenVerborgh** [comment]
You can read more at:
- http://mochajs.org/#asynchronous-code (not the Promises part)
- http://mochajs.org/#asynchronous-hooks

**Comment by pabloem** [comment]
Fixed by 8aa3670.
